File: ui/src/app/model/Task.ts
import {User} from './User';
import {Project} from './Project';

export class Task {
   id: string;
   title: string;
   priority: number;
   startDate: Date;
   endDate: Date;
   summary: string;
   parentTask: Task;
   user: User;
   project: Project;

}

File: service/src/main/java/com/fsd/project/manager/service/dao/data/Task.java
package com.fsd.project.manager.service.dao.data;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@EqualsAndHashCode
@Data
@Document(collection = "tasks")
public class Task {
    @Id
    private String id;
    @DBRef
    private Task parent;
    @DBRef
    private User user;
    @DBRef
    private Project project;
    private @Getter
    @Setter
    String title;
    private int priority;
    private String startDate;
    private String endDate;

}
